openhantek/src/hantekdso/dsoloop.cpp

```diff
@@ -380,7 +381,7 @@ void DsoLoop::convertRawDataToSamples(const std::vector<unsigned char> &rawData)
             for (unsigned pos = 0; pos < samples.size(); ++pos, bufferPosition += m_specification->channels) {
                 if (bufferPosition >= totalSampleCount) bufferPosition %= totalSampleCount;
                 const int16_t v = rawData[bufferPosition] - shiftDataBuf;
-                double samplePoint = (v / limit - harwareOffset) * gainStep - offsetCorrection;
+                double samplePoint = (v / limit - (harwareOffset+1)/2) * gainStep - offsetCorrection;
```

Comment:
v and limit are signed numbers, therefore v/limit is in range [0,1], while harwareOffset is in [-1,1].